src/node.cc

```diff
@@ -2867,9 +2869,54 @@ static bool ExecuteBootstrapper(Environment* env, Local<Function> bootstrapper,
 }
 
 
+void PromiseRejectCallback(PromiseRejectMessage message) {
+  Local<Promise> promise = message.GetPromise();
+  Isolate* isolate = promise->GetIsolate();
+  PromiseRejectEvent event = message.GetEvent();
+
+  Environment* env = Environment::GetCurrent(isolate);
+  Local<Function> callback;
+  Local<Value> value;
+
+  if (event == v8::kPromiseRejectWithNoHandler) {
+    callback = env->promise_reject_unhandled_function();
+    value = message.GetValue();
+
+    if (value.IsEmpty())
+      value = Undefined(isolate);
+  } else if (event == v8::kPromiseHandlerAddedAfterReject) {
+    callback = env->promise_reject_handled_function();
+    value = Undefined(isolate);
+  } else {
+    UNREACHABLE();
+  }
+
+  Local<Value> args[] = { promise, value };
+  MaybeLocal<Value> ret = callback->Call(env->context(),
+                                         Undefined(isolate),
+                                         arraysize(args),
+                                         args);
+
+  if (!ret.IsEmpty() && ret.ToLocalChecked()->IsTrue())
+    env->tick_info()->promise_rejections_toggle_on();
+}
+
 void LoadEnvironment(Environment* env) {
+  LoadEnvironmentGeneric(env, true, true);
+}
+
+void LoadEnvironmentGeneric(Environment* env, bool standalone_mode,
+                            bool upstream_node_mode) {
```

Comment:
This API is arguably too specific to Electron.  Something like `LoadEnvironment(Environment* env, unsigned flags)` with flags to control individual hooks would be better.

Rather than say "I want standalone mode", you'd say "I want a) message listener hooks, and b) disable stdio inheritance."